frontend/src/components/NotificationsList/NotificationsList.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect, useState } from "react";
+import {
+  getNotifications,
+  markNotificationAsRead,
+} from "../../api/notifications";
+import styles from "./NotificationsList.module.css";
+
+const NotificationsList: React.FC = () => {
+  const [notifications, setNotifications] = useState<NotificationItem[]>([]);
+
+  useEffect(() => {
+    fetchNotifications();
+  }, []);
+
+  const fetchNotifications = async () => {
+    try {
+      const fetchedNotifications = await getNotifications();
+      setNotifications(fetchedNotifications);
+    } catch (error) {
+      console.error("Failed to fetch notifications:", error);
+    }
+  };
+
+  const handleMarkAsRead = async (notificationId: number) => {
+    try {
+      await markNotificationAsRead(notificationId);
+      setNotifications(
+        notifications.map((notif) =>
+          notif.id === notificationId ? { ...notif, status: "read" } : notif
+        )
+      );
+    } catch (error) {
+      console.error("Failed to mark notification as read:", error);
+    }
+  };
+
+  return (
+    <div className={styles.notificationsContainer}>
+      <h2>Notifications</h2>
+      {notifications.length === 0 ? (
+        <p>No notifications</p>
+      ) : (
+        <ul className={styles.notificationsList}>
+          {notifications.map((notification) => (
+            <li key={notification.id} className={styles.notificationItem}>
+              <p>{notification.content}</p>
+              <small>{new Date(notification.createdAt).toLocaleString()}</small>
+              {notification.status !== "read" && (
+                <button onClick={() => handleMarkAsRead(notification.id)}>
+                  Mark as read
```

Comment:
I think we shouldn't let users mark notifications as read. Especially if users come back after a long time and they might have a long list of notifications and it would be quite burdensome for them to mark every single notif as read. As long as the notif is rendered and is viewable to users, we should automatically mark them as read.

Reply:
I got it, I will tackle this when I am working on stretch features.